### PR TITLE
Hive monad minor improvements

### PR DESCRIPTION
**Pull request by stephanh**
* Create table can now handle relative paths
* Adds method to get on disk location for a table to Hive.

### CODE REVIEW

File: hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala [hunk 1]
```diff
@@ -188,6 +188,10 @@ object Hive {
   def getClient: Hive[IMetaStoreClient] =
     Hive((_, client) => Result.ok(client))
 
+  /** Gets the Hive conf and client. */
+  def getConfClient: Hive[(HiveConf, IMetaStoreClient)] =
```
Comment (SamRoberts) [inline hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala]: Do you expect this to be used often? I'm just wondering whether it's really worth having a specialized method just for this, as opposes to using zipped or something on `getClient` and a `getConf` method.

File: hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala [hunk 2]
```diff
@@ -340,6 +345,17 @@ object Hive {
...
   })
 
+  /** Gets the on disk location of a Hive table. */
+  def getPath(database: String, table: String): Hive[Path] = Hive((conf, client) =>
+    try {
+      val location = client.getTable(database, table).getSd.getLocation
+      Result.ok(FileSystem.get(conf).makeQualified(new Path(location)))
+    } catch {
+      case _: NoSuchObjectException => Result.fail(s"Table $database.$table does not exist")
+      case NonFatal(t)              => Result.error(s"Failed to check strict existence of $database.$table", t)
```
Comment (SamRoberts) [inline hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala]: this seems like an oddly specific error message for such a generic catch statement. Are you sure this is the only reason to throw an exception other than NoSuchObjectException?

Comment (stephanh) [inline hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala]: If you are referring to the second error message than it is wrong (I got it from another method). I pushed the correct error message.

File: hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala (hunk 1 above)
Comment (stephanh) [inline hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala]: Probably not but zipping the two together is also a bit cumbersome.

Are you concerned about the API blowing out?

Comment (SamRoberts) [inline hive/src/main/scala/au/com/cba/omnia/ebenezer/scrooge/hive/Hive.scala]: Not really concerned about this specific method all by itself, but if you were to apply the same philosophy all the time you would blow the API out. It's a minor thing, and now you've fixed the error message below I'll happily approve, but personally I would be using two separate getConf and getClient methods here.

**Comment by SamRoberts** [comment]
:+1:

**Pull request merged**